lib/internal/util.js

```diff
@@ -366,6 +366,43 @@ function spliceOne(list, index) {
   list.pop();
 }
 
+let getStructuredStack;
+function isInsideNodeModules() {
+  if (!getStructuredStack) {
+    // Lazy-load to avoid a circular dependency.
+    const { runInNewContext } = require('vm');
+    // Use `runInNewContext()` to get something tamper-proof and
+    // side-effect-free. Since this is currently only used for a deprecated API,
+    // the perf implications should be okay.
```

Comment:
I still suggest bailing out after 10k attempts (which are about ~0.1 second).

Reply:
@ChALkeR I think it might be helpful if you could explain why you are both very much pro-deprecation and in favor of moving *all* code off the deprecated API, but at the same time want to keep the performance on the same level as it was before?

Like @mafintosh said, isn't the perf hit a *good* incentive to move away?

Reply:
@addaleax Because, according to the numbers you presented, this change has the potential of silently adding a 100x synchronous performance hit on some workloads (for buffers of size 1 KiB), which could affect streams and stuff like that significantly.

Presenting that performance hit to people without any warning (so they have no idea what's going on) could be pretty disruptive and can cause issues with real deployments out there.

In my opinion, a deprecation warning would be less disruptive than a 100x silent performance hit.